### ISSUE TITLE
Calendar Validation: Show message for max selections and add screen reader tags for validation error

### ISSUE THREAD

**Issue by narin** [department-of-veterans-affairs/va.gov-team#5992]
## User Story or Problem Statement
As a veteran using a screen reader, I need validation errors to be read out so I can know how to proceed.

If 3 selections are already made, other checkboxes should have a disabled style.

Link to Slack convo: https://dsva.slack.com/archives/CMNQT72LX/p1582063242235500?thread_ts=1582060436.232100&cid=CMNQT72LX

## Acceptance Criteria
- [ ] Add `aria` tags to calendar validation error
- [ ] Other checkboxes should show as `disabled` if 3 are already selected

**Comment by narin** [comment]
@lenaecb @markgreenburg @rtwell

In the [slack convo](https://dsva.slack.com/archives/CMNQT72LX/p1582063242235500?thread_ts=1582060436.232100&cid=CMNQT72LX), @1Copenut said: "I will defer the design of this message, but think we could either add it as an appended text block to the “You can choose up to 3 dates. A scheduling coordinator will call you to schedule the best time for your appointment.” text, or maybe in its own block. It’s not an error, but more of a heads up."

Just wondering how we should display/focus this messaging.  We can display the typical red validation error text, or just `focus()` on the the "You can choose up to 3 dates..." message at the top, which I assume would trigger the screen reader to re-read it? @1Copenut can you confirm?

Here is a screenshot of the calendar for your reference:

![image](https://user-images.githubusercontent.com/786704/75270265-1e737d80-57af-11ea-9e60-cfc87870599c.png)

**Comment by rtwell** [comment]
what do you think about something like this:

![image](https://user-images.githubusercontent.com/12564977/75273560-1c211b80-57cf-11ea-9123-f351fcd53513.png)

**Comment by 1Copenut** [comment]
I agree with @rtwell that we should throw an error. Looking at this, I think we should also add SR-only text to the cells that have date(s) selected. Here's what I'm thinking, markup-wise:

**Error handler:**
```javascript
<div class="vaos-calendar__options-container">
  { !maxSelected ? (
  <fieldset>
    <legend class="vads-u-visibility--screen-reader">Please select an option for this date</legend>
    <div class="vaos-calendar__options vads-u-justify-content--flex-end">
      <div class="vaos-calendar__option-cell">
        <div class="vaos-calendar__option">
          <input id="checkbox-2020-03-13_0" name="optionTime" type="checkbox" value="AM">
          <label class="vads-u-margin--0 vads-u-font-weight--bold vads-u-color--primary" for="checkbox-2020-03-13_0">AM</label>
        </div>
      </div>
      ...
    </div>
  </fieldset>) : (
  <p class="" role="alert">You can only select up to three dates</p>) }
</div>
```

**Selected dates:**
```html
<div class="vaos-calendar__indicator-bubbles-container">
  <div class="vaos-calendar__indicator-bubble vads-u-border--2px vads-u-border-color--white vads-u-background-color--base">
    <!-- Hide the short (visual) from screen readers, add an SR-only block of text
    <span aria-hidden="true">PM</span>
    <span class="vads-u-visibility--screen-reader">PM date selected</span>
  </div>
</div>
```

**Comment by narin** [comment]
Thanks for the input!

@1Copenut Sounds good. We can add the sr-only text easily. 

@rtwell I like the copy, but having the error display inline with the cell would take a fair amount of lift.  Can we display it in the same area as the other validation message? Keep in mind we would automatically scroll the user up to it:

![image](https://user-images.githubusercontent.com/786704/75283132-158ea600-57c7-11ea-9b56-3becf1adfb4e.png)

**Comment by 1Copenut** [comment]
@narin I'll hold adding any comments until I see the error message in action. I have a hunch any feedback I offer will be off-base without the box + screen reader running. Same for the input(s) on the open cell that could/would be disabled.

**Comment by markgreenburg** [comment]
@peggygannon would be great if you could provide guidance on content for the validation message on the calendar widget!

**Comment by peggygannon** [comment]
@rtwell @markgreenburg @lenaecb Here's my content recommendation. It looks like this validation error will be placed at top of calendar: 

You can choose only up to 3 dates for your appointment.

**Comment by lenaecb** [comment]
Thanks @peggygannon - I agree with the content recommendation "You can only choose up to 3 dates for your appointment." cc: @narin 

Also @narin - Given the potential lift needed for showing inline, I think showing the error with the validation message still allows for prominence. I recommend we move forward with that approach.

We can revisit the work for inline error message future work.

**Comment by narin** [comment]
Thanks for the input all!

**Comment by lenaecb** [comment]
@narin - We good to close this?

**Comment by 1Copenut** [comment]
@lenaecb I'll review this one and #6241 in staging quick with NVDA/JAWS and close them out.

**Comment by narin** [comment]
@lenaecb Need to make a tiny suggested change, get it passed through Jenkins, merge and validate on staging before we can close this.

**Comment by narin** [comment]
Merged and validated in staging

**Comment by 1Copenut** [comment]
Confirmed in Staging with NVDA and JAWS

/cc @lenaecb
